src/hooks/useOnboardingLayout.ts

```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line no-restricted-imports
+import {useWindowDimensions} from 'react-native';
+import variables from '@styles/variables';
+
+type OnboardingLayout = {
+    shouldUseNarrowLayout: boolean;
+};
+
+/**
+ * Onboarding layout for medium screen width is narrowed similarly as on web/desktop.
+ */
+export default function useOnboardingLayout(): OnboardingLayout {
+    const {width: windowWidth} = useWindowDimensions();
+
+    return {shouldUseNarrowLayout: windowWidth > variables.mobileResponsiveWidthBreakpoint};
```

Comment:
We can use `isMediumScreenWidth` from `useWindowDimensions`

Reply:
`isMediumScreen` is limited has it's max width limited to mobile widths only:
`const isMediumScreenWidth = windowWidth > variables.mobileResponsiveWidthBreakpoint && windowWidth <= variables.tabletResponsiveWidthBreakpoint;`

But happy to accept other suggestions to improve this.

Reply:
Oh, I just got feedback from @mateuuszzzzz that the other difference is that `useOnboardingLayout` is only based on the screen width, while `useWindowDimensions` is hardcoded for native platforms to always show `isSmallScreenWidth`. We don't want that here.

Reply:
For now I added explanation comments above the hook, let me know if this is enough or if you have other proposals.

```
/**
 * The main difference between useOnboardingLayout and useWindowDimension is that
 * useWindowDimension hardcodes isSmallScreenWidth, isMediumScreenWidth and
 * isLargeScreenWidth on native platforms, while this hook below always takes
 * screen width into consideration, no matter the platform.
 */
```

Reply:
maybe we change the name from `shouldUseNarrowLayout`? we already have that in `useResponsiveLayout` i thought the name would conflict with that.

Reply:
Hmm I can't come up with a better name though 🙈

`shouldUseNarrowLayout` makes sense though as it is consistent with `useResponsiveLayout` and those two hooks can even be replaced easily and the `useResponsiveLayout` will still work, not that anyone would want to do it too often. Happy to change though if you have some ideas.